sagemaker-pipeline-compare-model-versions/notebook.ipynb

```diff
@@ -0,0 +1,703 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from sagemaker.inputs import TrainingInput\n",
+    "from sagemaker.workflow.steps import TrainingStep\n",
+    "from sagemaker.estimator import Estimator\n",
+    "\n",
+    "# Where to store the trained model\n",
+    "model_path = f\"s3://{bucket}/{prefix}/churn\"\n",
+    "\n",
+    "# Fetch container to use for training\n",
+    "image_uri = sagemaker.image_uris.retrieve(\n",
+    "    framework=\"xgboost\",\n",
+    "    region=region,\n",
+    "    version=\"1.2-2\",\n",
+    "    py_version=\"py3\",\n",
+    "    instance_type=training_instance_type,\n",
+    ")\n",
+    "\n",
+    "# Create XGBoost estimator object\n",
+    "# The object contains information about what container to use, what instance type etc.\n",
+    "xgb_estimator = Estimator(\n",
+    "    image_uri=image_uri,\n",
+    "    instance_type=training_instance_type,\n",
+    "    instance_count=1,\n",
+    "    output_path=model_path,\n",
+    "    role=role,\n",
+    "    disable_profiler=True,\n",
+    ")\n",
+    "\n",
+    "xgb_estimator.set_hyperparameters(\n",
+    "    max_depth=5,\n",
+    "    eta=0.2,\n",
+    "    gamma=4,\n",
+    "    min_child_weight=6,\n",
+    "    subsample=0.8,\n",
+    "    objective=\"binary:logistic\",\n",
+    "    num_round=25,\n",
+    ")\n",
+    "\n",
+    "# Use the xgb_estimator in a Sagemaker pipelines ProcessingStep.\n",
+    "# NOTE how the input to the training job directly references the output of the previous step.\n",
+    "step_train_model = TrainingStep(\n",
+    "    name=\"Train-Churn-Model\",\n",
+    "    estimator=xgb_estimator,\n",
+    "    inputs={\n",
+    "        \"train\": TrainingInput(\n",
+    "            s3_data=step_preprocess_data.properties.ProcessingOutputConfig.Outputs[\n",
+    "                \"train\"\n",
+    "            ].S3Output.S3Uri,\n",
+    "            content_type=\"text/csv\",\n",
+    "        ),\n",
+    "        \"validation\": TrainingInput(\n",
+    "            s3_data=step_preprocess_data.properties.ProcessingOutputConfig.Outputs[\n",
+    "                \"validation\"\n",
+    "            ].S3Output.S3Uri,\n",
+    "            content_type=\"text/csv\",\n",
+    "        ),\n",
+    "    },\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<a id='evaluate'></a>\n",
+    "\n",
+    "## Evaluate model step\n",
+    "When a model a trained, it's common to run some sanity check on it before registering it with the model registry. This ensures the model registry isn't cluttered with poorly performing model versions. To evaluate the model, create a ScriptProcessor object and use it in a ProcessingStep.\n",
```

Comment:
just to get the wordings right, its not a sanity check. but measuring the performance of the model on unseen data. please rewrite the sentence